crates/cheatnet/src/execution/syscalls.rs

```diff
@@ -291,24 +290,64 @@ impl CheatableSyscallHandler<'_> {
             response_wrapper.write(vm, &mut self.syscall_handler.syscall_ptr)?;
             return Ok(());
         } else if SyscallSelector::CallContract == selector {
+            let StarknetHint::SystemCall { system: syscall } = hint else {
+                return Err(HintError::CustomHint(
+                    "Test functions are unsupported on starknet.".into(),
+                ));
+            };
+            let initial_syscall_ptr = get_ptr_from_res_operand_unchecked(vm, syscall);
+            self.verify_syscall_ptr(initial_syscall_ptr)?;
+
             // Increment, since the selector was peeked into before
             self.syscall_handler.syscall_ptr += 1;
+
+            if selector != SyscallSelector::Keccak {
+                self.syscall_handler
+                    .increment_syscall_count_by(&selector, 1);
+            }
             return self.execute_syscall(
                 vm,
                 call_contract_syscall,
                 constants::CALL_CONTRACT_GAS_COST,
             );
         } else if SyscallSelector::LibraryCall == selector {
+            let StarknetHint::SystemCall { system: syscall } = hint else {
+                return Err(HintError::CustomHint(
+                    "Test functions are unsupported on starknet.".into(),
+                ));
+            };
+            let initial_syscall_ptr = get_ptr_from_res_operand_unchecked(vm, syscall);
+            self.verify_syscall_ptr(initial_syscall_ptr)?;
+
             // Increment, since the selector was peeked into before
             self.syscall_handler.syscall_ptr += 1;
+
+            if selector != SyscallSelector::Keccak {
```

Comment:
The selector is already `!= Keccak` here

Reply:
https://github.com/foundry-rs/starknet-foundry/pull/680/commits/17bd90929784720a4bd1eb6692ce5108c409c005